### PR TITLE
Fixing scrolling

### PR DESCRIPTION
**Pull request by KamranAsif**
## Description
Fixed a bug why the scrolling wasn't always rendering correctly.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/reducers/scrollStateHelper.js [hunk 1]
```diff
@@ -154,39 +154,6 @@ function _addRowToBuffer(state, rowIndex, firstViewportRowIndex, lastViewportRow
 };
 
 /**
- * combine with updateRows with additioona prop
- *
- * @param {!Object} state
- * @private
- */
-function _updateRowsWithBuffer(state) {
```
Comment (KamranAsif) [inline src/reducers/scrollStateHelper.js]: Merged this with updateRows

File: src/reducers/scrollStateHelper.js [hunk 2]
```diff
@@ -437,9 +399,6 @@ export function scrollTo(state, scrollPosition) {
   _updateHeightsInViewport(state, firstRowIndex, firstRowOffset);
   _updateHeightsAboveViewport(state, firstRowIndex);
 
-  _updateRows(state);
-  _recalculateRowHeights(state);
-
   return Object.assign({}, state, {
```
Comment (KamranAsif) [inline src/reducers/scrollStateHelper.js]: The bug was this line was after updateRows. I fix it by refactoring the updateVisibleRows out of these functions

**Comment by KamranAsif** [comment]
Demo: http://kamranasif.github.io/fixed-data-table-2

File: src/reducers/scrollStateReducer.js [hunk 3]
```diff
@@ -48,10 +49,10 @@ function scrollStateReducer(state = DEFAULT_STATE, action) {
   switch (action.type) {
     case ActionTypes.INITIALIZE:
       let { props } = action;
-
       state = updateRowCount(state, props);
       state = updateRowHeights(state, props);
       state = updateViewHeight(state, props);
+      state = updateVisibleRows(state);
```
Comment (wcjordan) [inline src/reducers/scrollStateReducer.js]: I'm assuming these methods aren't independent.  Given that, what's a good way to keep track of their interdependencies?  Can we add a comment stating when updateVisibleRows needs to be called and what helper methods it needs to be called after and why.

**Comment by wcjordan** [comment]
lgtm

File: src/FixedDataTableRoot-test.js [hunk 4]
```diff
@@ -100,32 +105,65 @@ describe('FixedDataTableRoot', function() {
...
+      let table = renderTable({scrollTop: 600});
+      assert.equal(table.getTableState().scrollY, 600, 'should set scrollY to 600');
+
+      table = renderTable({scrollTop: 300});
+      assert.equal(table.getTableState().scrollY, 300, 'should set scrollY to 300');
+    });
 
+    it('should update scrollToColumn correctly', function() {
+      let table = renderTable({scrollToColumn: 3});
+      assert.equal(table.getTableState().scrollX, 300 * 2, 'should be third visible column');
```
Comment (wcjordan) [inline src/FixedDataTableRoot-test.js]: how does the mapping here work?  it's weird scrolling to the first is * 1 and scrolling to the 3rd is * 2.  Can you add a short comment on why that happens

Comment (KamranAsif) [inline src/FixedDataTableRoot-test.js]: Should be 3. Not sure why it was passing

**Comment by bluejamesbond** [comment]
@KamranAsif Just wanted to see if this is trying to address the scenario: http://kamranasif.github.io/fixed-data-table-2/example-collapse-rows.html

Scrolling all the way to the bottom row and then expanding it. However, the scroller never updates to include the additional height.

**Comment by wcjordan** [comment]
@KamranAsif  can we get this merged in?  Let me know if any conflicts are hard to address, and I can pair through them.

**Comment by KamranAsif** [comment]
Hey,

The only issue was the travis check failing. We can merge it in as is and let the test fail if you want

**Comment by wcjordan** [comment]
Let's merge and I can look at fixing tests separately.  I believe they've been broken on the branch since redux was introduced, although this break appears to be something around how we're using lodash.

**Pull request merged**